hls-graph/src/Development/IDE/Graph/Internal/Database.hs

```diff
@@ -303,6 +314,7 @@ mapConcurrentlyAIO_ _ [] = pure ()
 mapConcurrentlyAIO_ f [one] = liftIO $ justWait $ fmap f one
 mapConcurrentlyAIO_ f many = do
     ref <- AIO ask
+    -- mask to make sure we keep track of all the asyncs
     waits <- liftIO $ uninterruptibleMask $ \restore -> do
         waits <- liftIO $ traverse (waitOrSpawn . fmap (restore . f)) many
```

Comment:
This `restore` is potentially running in a different thread, as `waitOrSpawn` calls `async`. I wonder what the semantics of `restore` are when invoked in a different thread ...

Reply:
I think it is the expected semantics, since threads inherit their masking state from the parent.